### PR TITLE
feat: add fallback dialing support

### PR DESCRIPTION
**Pull request by dovholuknf**
Should core-keeper require overriding the dialer this is where/how I think it makes sense.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]
please fix the unit test and gen the mock for the interface change

File: bootstrap/secret/jwtprovider.go [hunk 1]
```diff
@@ -14,23 +14,15 @@ import (
...
+	secretProvider      interfaces.SecretProviderExt
+	defaultRoundTripper http.RoundTripper
 }
 
 func NewJWTSecretProvider(secretProvider interfaces.SecretProviderExt) clientinterfaces.AuthenticationInjector {
 	return &jwtSecretProvider{
 		secretProvider: secretProvider,
 	}
 }
-func NewJWTSecretProviderWithRT(secretProvider interfaces.SecretProviderExt, roundTripper_b http.RoundTripper) clientinterfaces.AuthenticationInjector {
```
Comment (cloudxxx8) [inline bootstrap/secret/jwtprovider.go:26]: will this removeal break anything?

Comment (dovholuknf) [reply bootstrap/secret/jwtprovider.go:26]: [I searched acrosss the whole org with github for the function but found nothing ](https://github.com/search?q=org%3Aedgexfoundry%20NewJWTSecretProviderWithRT&type=code)using it. I was surprised to see an unused function so I removed it. tbh it sort of looked like it was the nascent attempt at providing fallback support.

Comment (cloudxxx8) [reply bootstrap/secret/jwtprovider.go:26]: ok, thanks

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**